### PR TITLE
abstraction-chat-filtering-project

### PR DESCRIPTION
**Pull request by kylmrcdr123**

### CODE REVIEW

File: 04 Abstraction Exercises Part 1/abstraction-chat-filtering-project/src/main/java/Main.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+                    return "Take care always.";
+                case 4:
+                    return "Congrats";
+                default:
+            }
+        }
+        return null;
+    }
+
+    private static boolean containsOffensiveWord(String message) {
```
Comment (RoCS2024) [inline 04 Abstraction Exercises Part 1/abstraction-chat-filtering-project/src/main/java/Main.java:90]: Do not use regex for comparing the input to the list of bad words.

File: 04 Abstraction Exercises Part 1/abstraction-chat-filtering-project/src/main/java/Chat.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+import java.util.ArrayList;
+        import java.util.List;
+
+public class Chat {
+    private String senderName;
+    private String recipientName;
+    private List<String> messages;
+    private boolean recipientOnline;
+    private List<String> bannedWords;
```
Comment (RoCS2024) [inline 04 Abstraction Exercises Part 1/abstraction-chat-filtering-project/src/main/java/Chat.java:13]: this should be used in all the instances of the Chat class

File: 04 Abstraction Exercises Part 1/abstraction-chat-filtering-project/src/main/java/Main.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        do {
+            System.out.println("Choose from the options: ");
+            System.out.println("1. Status");
+            System.out.println("2. Type message");
+            System.out.println("0. Back");
+            option = scanner.nextInt();
+
+            switch (option) {
+                case 1:
+                    if (!chat.isRecipientOnline()) {
```
Comment (RoCS2024) [inline 04 Abstraction Exercises Part 1/abstraction-chat-filtering-project/src/main/java/Main.java:26]: what if the user does not want to be online?

File: 04 Abstraction Exercises Part 1/abstraction-chat-filtering-project/src/main/java/Main.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        System.out.println("Choose a message to send:");
+        System.out.println("1. \"Good morning!\"");
+        System.out.println("2. \"Hello!\"");
+        System.out.println("3. \"Take care always\"");
+        System.out.println("4. \"Congrats\"");
+        System.out.println("5. \"Type your message:\"");
+
+        int choice;
+        do {
+            System.out.print("Enter your choice: ");
```
Comment (RoCS2024) [inline 04 Abstraction Exercises Part 1/abstraction-chat-filtering-project/src/main/java/Main.java:62]: any validation why the loop has to go on?

**Review by RoCS2024 — CHANGES_REQUESTED** [review]
see comments

File: 04 Abstraction Exercises Part 1/abstraction-chat-filtering-project/src/main/java/Chat.java (hunk 2 above)
Comment (kylmrcdr123) [reply 04 Abstraction Exercises Part 1/abstraction-chat-filtering-project/src/main/java/Chat.java:13]: Noted sir

File: 04 Abstraction Exercises Part 1/abstraction-chat-filtering-project/src/main/java/Main.java (hunk 1 above)
Comment (kylmrcdr123) [reply 04 Abstraction Exercises Part 1/abstraction-chat-filtering-project/src/main/java/Main.java:90]: Noted sir

**Pull request closed without merging**